assets/js/app.js

```diff
@@ -222,9 +222,13 @@ function setupRoutes(map, pos, targetRouteLength) {
         pos.lat + KM_RADIUS_TO_SEARCH_FOR_ROUTES_NEAR,
         pos.lng + KM_RADIUS_TO_SEARCH_FOR_ROUTES_NEAR,
         function(data) {
-            var walkRoute = getRandomWalkFromOsmDataset(data, targetRouteLength);
-
-            renderWalk(walkRoute, pos);
+            if (hashVal) {
+                var walkRoute = getWalkFromOsmDatasetById(data, hashVal.split('#')[1]);
```

Comment:
Hmm.. I can see a problem here in that if you aren't located in the same radius as your friend the walk id may not be in the OSM dataset - eg. if I'm on the gold coast and you send me a walk in brisbane that's not within 20km of me I won't see it :(

We might have to query OSM directly for the specific walk id using a different api query to make sure we get the right id